lib/libspl/Makefile.am

```diff
@@ -1,7 +1,7 @@
 include $(srcdir)/%D%/include/Makefile.am
 
 libspl_assert_la_CFLAGS = $(AM_CFLAGS) $(LIBRARY_CFLAGS) $(LIBUNWIND_CFLAGS)
-libspl_la_CFLAGS        = $(libspl_assert_la_CFLAGS)
+libspl_la_CFLAGS        = $(libspl_assert_la_CFLAGS) -Wno-error=atomic-alignment
```

Comment:
This is causing a build failure.  At a minimum `config/always-compiler-options.m4` is going to need to be updated to check if the option exists.

```
  make[4]: Entering directory '/tmp/zfs-build-zfs-cvMTsZNL/BUILD/zfs-2.4.99-build/zfs-2.4.99'
    CC       module/zfs/libicp_la-zfs_impl.lo
    CC       lib/libspl/libspl_la-atomic.lo
    CC       lib/libspl/libspl_la-getexecname.lo
    CC       lib/libspl/libspl_la-list.lo
  cc1: error: '-Wno-error=atomic-alignment': no option '-Watomic-alignment'
  cc1: error: '-Wno-error=atomic-alignment': no option '-Watomic-alignment'
  make[4]: *** [Makefile:9338: lib/libspl/libspl_la-getexecname.lo] Error 1
```